pkg/sql/logictest/tests/local-mixed-24.1/local-mixed-24_1_test: TestLogic_union failed

pkg/sql/logictest/tests/local-mixed-24.1/local-mixed-24_1_test.TestLogic_union [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17023978?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Ci_TestsAwsLinuxArm64_UnitTests/17023978?buildTab=artifacts#/) on master @ [25a8796e44293214a188a8e2e852a3084e7dbb8e](https://github.com/cockroachdb/cockroach/commits/25a8796e44293214a188a8e2e852a3084e7dbb8e):


```
=== RUN   TestLogic_union
    test_log_scope.go:170: test logs captured to: /artifacts/tmp/_tmp/2bb97c65bfc8d34248211a050245bd97/logTestLogic_union372032161
    test_log_scope.go:81: use -show-logs to present logs inline
    test_server_shim.go:144: cluster virtualization disabled in global scope due to issue: #76378 (expected label: C-bug)
    logic.go:3032: 
         pq: txn already encountered an error; cannot be used anymore (previous err: aborted in DistSender: result is ambiguous: context canceled)
[21:40:34] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/14365/execroot/com_github_cockroachdb_cockroach/bazel-out/aarch64-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.1/local-mixed-24_1_test_/local-mixed-24_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/union with config local-mixed-24.1: 102 tests, 1 failures
[21:40:35] --- total progress: 102 statements
--- total: 102 tests, 1 failures
    logic.go:4391: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/2bb97c65bfc8d34248211a050245bd97/logTestLogic_union372032161
--- FAIL: TestLogic_union (3.32s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_union.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

pkg/sql/logictest/tests/local-mixed-24.1/local-mixed-24_1_test.TestLogic_union [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/17110890?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/17110890?buildTab=artifacts#/) on master @ [74333311616b937fea6a995462215a1cb5962686](https://github.com/cockroachdb/cockroach/commits/74333311616b937fea6a995462215a1cb5962686):


```
[07:04:18] CREATE TABLE abcd (a INT, b INT, c INT, d INT, INDEX (a, b, c DESC) STORING (d));
INSERT INTO abcd VALUES (1, 1, 1, 1), (1, 2, 2, 2), (1, 2, 3, 3), (2, 2, 2, 2), (2, 2, 3, 3);
CREATE TABLE efgh (e INT PRIMARY KEY, f INT, g INT, h INT);
INSERT INTO efgh VALUES (1, 1, 1, 1), (2, 2, 2, 2), (3, 2, 3, 3), (4, 2, 2, 2), (5, 2, 3, 3);;
[07:04:18] rewrote:
CREATE TABLE abcd (a INT8, b INT8, c INT8, d INT8, INDEX (a, b, c DESC) STORING (d), FAMILY (a), FAMILY (d), FAMILY (b), FAMILY (c));
INSERT INTO abcd VALUES (1, 1, 1, 1), (1, 2, 2, 2), (1, 2, 3, 3), (2, 2, 2, 2), (2, 2, 3, 3);
CREATE TABLE efgh (e INT8 PRIMARY KEY, f INT8, g INT8, h INT8, FAMILY (e, g, h), FAMILY (f));
INSERT INTO efgh VALUES (1, 1, 1, 1), (2, 2, 2, 2), (3, 2, 3, 3), (4, 2, 2, 2), (5, 2, 3, 3);


[07:04:18] SELECT e, f, g, h FROM efgh EXCEPT SELECT a, b, c, d FROM abcd ORDER by e, f;
[07:04:18] 	-- OK;
[07:04:18] SELECT a, b AS b1, b AS b2 FROM abc EXCEPT SELECT a, b, c FROM abc ORDER by a, b1;
[07:04:18] 	-- OK;
[07:04:18] SELECT a, b AS b1, b AS b2 FROM abc INTERSECT SELECT a, c, b FROM abc ORDER by a, b2;
[07:04:18] 	-- OK;
[07:04:18] CREATE TABLE t127043_1 (k1 INT, v1 INT, INDEX (k1));
INSERT INTO t127043_1 VALUES (1, 1);
CREATE TABLE t127043_2 (k2 INT, v2 INT, INDEX (k2));
INSERT INTO t127043_2 VALUES (1, 1);
CREATE TABLE t127043_3 (k3 INT, v3 INT, INDEX (k3));
INSERT INTO t127043_3 VALUES (1, 1);
CREATE VIEW v127043 (k, v) AS
  SELECT
    k1 AS k, v1 AS v FROM t127043_1@t127043_1_k1_idx
  UNION SELECT
    k2 AS k, v2 AS v FROM t127043_2@t127043_2_k2_idx
  UNION SELECT
    k3 AS k, v3 AS v FROM t127043_3@t127043_3_k3_idx;;
[07:04:18] rewrote:
CREATE TABLE t127043_1 (k1 INT8, v1 INT8, INDEX (k1), FAMILY (k1), FAMILY (v1));
INSERT INTO t127043_1 VALUES (1, 1);
CREATE TABLE t127043_2 (k2 INT8, v2 INT8, INDEX (k2), FAMILY (v2, k2));
INSERT INTO t127043_2 VALUES (1, 1);
CREATE TABLE t127043_3 (k3 INT8, v3 INT8, INDEX (k3), FAMILY (v3), FAMILY (k3));
INSERT INTO t127043_3 VALUES (1, 1);
CREATE VIEW v127043 (k, v) AS SELECT k1 AS k, v1 AS v FROM t127043_1@t127043_1_k1_idx UNION SELECT k2 AS k, v2 AS v FROM t127043_2@t127043_2_k2_idx UNION SELECT k3 AS k, v3 AS v FROM t127043_3@t127043_3_k3_idx;


[07:04:18] SELECT k, v FROM v127043 WHERE k = 1 LIMIT 1;;
[07:04:18] 	-- FAIL
    logic.go:3032: 
         pq: txn already encountered an error; cannot be used anymore (previous err: aborted in DistSender: result is ambiguous: context canceled)
[07:04:18] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/3457/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/local-mixed-24.1/local-mixed-24_1_test_/local-mixed-24_1_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/union with config local-mixed-24.1: 102 tests, 1 failures
[07:04:21] --- total progress: 102 statements
--- total: 102 tests, 1 failures
    logic.go:4391: -- test log scope end --
test logs left over in: /artifacts/tmp/_tmp/584afdc678f8faf3da53757d63161a89/logTestLogic_union3256016657
--- FAIL: TestLogic_union (7.55s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_union.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)
</sub>

--- a/pkg/sql/logictest/testdata/logic_test/union
+++ b/pkg/sql/logictest/testdata/logic_test/union
@@ -683,11 +683,15 @@ SELECT a, b AS b1, b AS b2 FROM abc INTERSECT SELECT a, c, b FROM abc ORDER by a
 # synchronizer was causing spurious errors on queries with LIMIT.
 statement ok
 CREATE TABLE t127043_1 (k1 INT, v1 INT, INDEX (k1));
-INSERT INTO t127043_1 VALUES (1, 1);
 CREATE TABLE t127043_2 (k2 INT, v2 INT, INDEX (k2));
-INSERT INTO t127043_2 VALUES (1, 1);
 CREATE TABLE t127043_3 (k3 INT, v3 INT, INDEX (k3));
+
+statement ok
+INSERT INTO t127043_1 VALUES (1, 1);
+INSERT INTO t127043_2 VALUES (1, 1);
 INSERT INTO t127043_3 VALUES (1, 1);
+
+statement ok
 CREATE VIEW v127043 (k, v) AS
   SELECT
     k1 AS k, v1 AS v FROM t127043_1@t127043_1_k1_idx

